verify_rpms/rpm_verifier.py

```diff
@@ -200,28 +289,41 @@ def __call__(self, img: str) -> ProcessedImage:
     required=True,
 )
 def main(
-    img_input: str,
+    image_url: str,
+    image_digest: str,
     fail_unsigned: bool,
     workdir: Path,
 ) -> None:
     """Verify RPMs are signed"""
     status_path: Path = workdir / "status"
-    results_path: Path = workdir / "results"
+
+    # Exit in case of failure to process the image reference
+    try:
+        process = inspect_image_ref(image_url=image_url, image_digest=image_digest)
+        images = get_images_from_inspection(
+            inspect_results=process, image_url=image_url, image_digest=image_digest
+        )
+    except CalledProcessError as err:
+        print(f"failed to process image\n{err.stderr}")
+        status_path.write_text("ERROR")
+        if fail_unsigned:
+            sys.exit(f"failed to process image\n{err.stderr}")
+        else:
+            sys.exit(0)
 
     processor = ImageProcessor(workdir=workdir)
-    processed_image = processor(img=img_input)
+    with ThreadPoolExecutor() as executor:
+        processed_images: Iterable[ProcessedImage] = executor.map(processor, images)
+
+    output, failures_occurred = set_output_and_status(list(processed_images))
 
-    failures_occurred, output = generate_output(processed_image=processed_image)
-    results = generate_results(processed_image=processed_image)
-    results_path.write_text(json.dumps(results))
     if failures_occurred:
         status_path.write_text("ERROR")
     else:
         status_path.write_text("SUCCESS")
     if failures_occurred and fail_unsigned:
-        sys.exit(output + "\n" + json.dumps(results))
+        sys.exit(output)
```

Comment:
it will be included in the next PR:
https://github.com/redhat-appstudio/tools/pull/240/files#diff-55de11f901dfeb0d3386ef50dff76dbcaece97db7d2c9dfaff02448237fbbb6cR357